src/awx_plugins/credentials/plugins.py

```diff
@@ -532,6 +534,100 @@
+            },
+            {
+                'id': 'app_id',
+                'label': 'Github App id created by the GitHub Admin',
+                'type': 'string',
+                'help_text': gettext_noop(
+                    'The GitHub App Id created by the GitHub Admin '
+                    'Example App ID: 1121547 '
+                    'from https://github.com/settings/apps/arestlel-github-app-ansible'
+                    'required for creating a bearer token for authentication',
+                ),
+            },
+            {
+                'id': 'install_id',
+                'label': 'Installation id',
+                'type': 'string',
+                'help_text': gettext_noop(
+                    'The  Installation ID from the GitHub App installation generated by the GitHub Admin '
+                    'Example: 59980338 '
+                    'he  Installation ID from the GitHub App installation generated by the GitHub Admin '
+                    'Example: 59980338 '
+                    'extracted from the installation link: https://github.com/settings/installations/59980338 '
+                    'required for creating a limited github app token'
+                ),
+            },                      
+            {
+                'id': 'ssh_key_data',
+                'label': gettext_noop('RSA Private Key'),
+                'type': 'string',
+                'format': 'ssh_private_key',
+                'secret': True,
+                'multiline': True,
+                'help_text': gettext_noop(
+                    'Paste the contents of the PEM file that the GitHub Admin '
+                    'provided to you with the app and installation ids.',
+                ),
+            },
+        ],
+        'required': ['install_id','ssh_key_data',],
+    },
+)
+
+
+def github_app_backend(**kwargs):
+    github_url = kwargs.get("github_url")
+    app_id = kwargs.get("app_id")
+    installation_id = int(kwargs.get("install_id"))
+    private_key = kwargs.get("ssh_key_data")
+    jwt_expiry = int(kwargs.get("jwt_expiry", 600))
+
+    missing_parameters = []
+    if not github_url:
+        missing_parameters.append("GitHub URL")
+    if not installation_id:
+        missing_parameters.append("Installation ID")
+    if not app_id:
+        missing_parameters.append("Application ID")
+    if not private_key:
+        missing_parameters.append("Private Key")
+    if missing_parameters:
+        raise Exception.MissingParameterError(missing_parameters)
+
+    auth = Auth.AppAuth(app_id=app_id, private_key=private_key, jwt_expiry=jwt_expiry).get_installation_auth(
+        installation_id=installation_id
+    )
+
+    # In order to generate a token we have to make an initial call
+    # to GitHub even though we dont need to interact with GitHub.
+    Github(auth=auth)
+
+    return auth.token
+
+github_app_plugin = CredentialPlugin("GitHub App Authentication", inputs=github_app_token, backend=github_app_backend)
```

Comment:
An entry for this would be needed in `[project.entry-points."awx_plugins.credentials"]` of `pyproject.toml`.